File: backend/app/services/company_profile_service.py
"""Company Profile service"""
from typing import Optional, Dict, Any
from sqlalchemy.ext.asyncio import AsyncSession
from sqlalchemy import select, and_
from app.models.company_profile import CompanyProfile


async def get_company_profile(
    db: AsyncSession,
    tenant_id: str,
) -> Optional[CompanyProfile]:
    """Get company profile for tenant"""
    result = await db.execute(
        select(CompanyProfile).where(CompanyProfile.tenant_id == tenant_id)
    )
    return result.scalar_one_or_none()


async def create_company_profile(
    db: AsyncSession,
    tenant_id: str,
    data: Dict[str, Any],
) -> CompanyProfile:
    """Create company profile"""
    profile = CompanyProfile(
        tenant_id=tenant_id,
        **data
    )
    db.add(profile)
    await db.commit()
    await db.refresh(profile)
    return profile


async def update_company_profile(
    db: AsyncSession,
    tenant_id: str,
    data: Dict[str, Any],
) -> Optional[CompanyProfile]:
    """Update company profile"""
    profile = await get_company_profile(db, tenant_id)
    if not profile:
        return None
    
    # Update fields
    for key, value in data.items():
        if value is not None:
            setattr(profile, key, value)
    
    await db.commit()
    await db.refresh(profile)
    return profile


async def get_company_context_for_proposals(
    db: AsyncSession,
    tenant_id: str,
) -> Dict[str, Any]:
    """Get company profile formatted for proposal generation"""
    profile = await get_company_profile(db, tenant_id)
    if not profile:
        return {}
    
    return {
        "company_name": profile.company_name,
        "legal_name": profile.legal_name,
        "company_overview": profile.company_overview,
        "mission_statement": profile.mission_statement,
        "core_values": profile.core_values or [],
        "differentiators": profile.differentiators or [],
        "core_capabilities": profile.core_capabilities or [],
        "technical_expertise": profile.technical_expertise or [],
        "service_offerings": profile.service_offerings or [],
        "certifications": profile.certifications or [],
        "contract_vehicles": profile.contract_vehicles or [],
        "past_performance_highlights": profile.past_performance_highlights or [],
        "key_personnel": profile.key_personnel or [],
        "win_themes": profile.win_themes or [],
        "standard_boilerplate": profile.standard_boilerplate or {},
    }

File: frontend/src/services/companyProfileService.ts
import api from './api'

export interface CompanyProfile {
  id: string
  tenant_id: string
  company_name: string
  legal_name?: string
  duns_number?: string
  cage_code?: string
  uei?: string
  website?: string
  email?: string
  phone?: string
  address?: string
  city?: string
  state?: string
  zip_code?: string
  country: string
  mission_statement?: string
  vision_statement?: string
  company_overview?: string
  core_values?: string[]
  differentiators?: string[]
  business_type?: string
  size_standard?: string
  naics_codes?: string[]
  sic_codes?: string[]
  contract_vehicles?: string[]
  certifications?: string[]
  security_clearances?: string[]
  compliance_frameworks?: string[]
  core_capabilities?: string[]
  technical_expertise?: string[]
  service_offerings?: string[]
  industry_experience?: string[]
  past_performance_highlights?: Array<{
    contract_name?: string
    agency?: string
    value?: number
    description?: string
    period?: string
  }>
  key_contracts?: Array<{
    contract_name?: string
    agency?: string
    value?: number
    description?: string
  }>
  awards_recognition?: Array<{
    award_name?: string
    organization?: string
    year?: string
    description?: string
  }>
  key_personnel?: Array<{
    name?: string
    title?: string
    expertise?: string
    bio?: string
  }>
  executive_team?: Array<{
    name?: string
    title?: string
    bio?: string
  }>
  standard_boilerplate?: Record<string, string>
  win_themes?: string[]
  proposal_templates?: Record<string, any>
  annual_revenue?: string
  number_of_employees?: string
  years_in_business?: number
  created_at: string
  updated_at: string
}

export const companyProfileService = {
  get: async (): Promise<CompanyProfile> => {
    const response = await api.get<CompanyProfile>('/company-profile')
    return response.data
  },

  create: async (data: Partial<CompanyProfile>): Promise<CompanyProfile> => {
    const response = await api.post<CompanyProfile>('/company-profile', data)
    return response.data
  },

  update: async (data: Partial<CompanyProfile>): Promise<CompanyProfile> => {
    const response = await api.put<CompanyProfile>('/company-profile', data)
    return response.data
  },

  getContext: async (): Promise<Record<string, any>> => {
    const response = await api.get<Record<string, any>>('/company-profile/context')
    return response.data
  },
}

